hkmc2/shared/src/main/scala/hkmc2/codegen/Lowering.scala

```diff
@@ -414,7 +414,13 @@ class Lowering()(using Config, TL, Raise, State, Ctx):
       raise:
         WarningReport(msg"Pure expression in statement position" -> t.toLoc :: Nil, S(t))
     
-    val insted = t.instantiated
+    @tailrec
+    def getAnnots(t: st, acc: List[Annot]): (List[Annot], st) = t match
+      case st.Annotated(annot, trm) => getAnnots(trm, annot :: acc)
+      case _ => (acc, t)
+    val (annots, trm) = getAnnots(t, Nil)
+    
+    val insted = trm.instantiated
```

Comment:
It should match on `insted`, rather than `trm`. Basically, one should never pattern match on terms before "isntantiating" them.
```suggestion
    def getAnnots(t: st, acc: List[Annot]): (List[Annot], st) = t match
      case st.Annotated(annot, trm) => getAnnots(trm.instantiated, annot :: acc)
      case _ => (acc, t)
    val insted = trm.instantiated
    val (annots, trm) = getAnnots(insted, Nil)
```